ingest/sqs/pools/transformer/pool_transformer.go

```diff
@@ -191,19 +194,68 @@ func (pi *poolTransformer) convertPool(
 	}()
 
 	balances := pi.bankKeeper.GetAllBalances(ctx, pool.GetAddress())
+
+	// Convert pool denoms to map for faster lookup.
+	poolDenomsMap := getPoolDenomsMap(pool.GetPoolDenoms(ctx))
+
+	spreadFactor := pool.GetSpreadFactor(ctx)
+
+	// Get pool denoms. Although these can be inferred from balances, this is safer.
+	// If we used balances, for pools with no liquidity, we would not be able to get the denoms.
+	denoms, err := pi.poolManagerKeeper.RouteGetPoolDenoms(ctx, pool.GetId())
+	if err != nil {
+		return nil, err
+	}
+
+	var cosmWasmPoolModel *sqsdomain.CosmWasmPoolModel
 	if pool.GetType() == poolmanagertypes.CosmWasm {
 		cwPool, ok := pool.(cosmwasmpooltypes.CosmWasmExtension)
 		if !ok {
 			return nil, fmt.Errorf("pool (%d) with type (%d) is not a CosmWasmExtension", pool.GetId(), pool.GetType())
 		}
 
 		balances = cwPool.GetTotalPoolLiquidity(ctx)
-	}
 
-	// Convert pool denoms to map for faster lookup.
-	poolDenomsMap := getPoolDenomsMap(pool.GetPoolDenoms(ctx))
+		// This must never happen, but if it does, and there is no checks, the query will fail silently
+		// so we panic here to make sure we catch this error
+		if pi.wasmKeeper == nil {
+			panic("wasmKeeper is nil")
+		}
 
-	spreadFactor := pool.GetSpreadFactor(ctx)
+		bz := pi.wasmKeeper.QueryRaw(ctx, cwPool.GetAddress(), []byte("contract_info"))
```

Comment:
Would it be possible to encapsulate the new logic and/or the CW-pool-specific logic into a separate function and cover by unit tests?